### PR TITLE
Add cloudwatch logging

### PR DESCRIPTION
**Pull request by dehort**
Added the request_id middleware to help with logging the request_id.
Moved the usage of the ReceptorServiceFactory into the handshake_handler.
Passing the logger instance through the websocket/receptor layer.  The API layer will get a different logger instance and request_id.

I think I need to get some miles on this and play around with the log messages in kibana.

### CODE REVIEW

**Comment by dehort** [comment]
@Jason-RH sorry this is a crazy huge PR (again) and really painful to review (again)...  :(

File: internal/controller/api/job_receiver.go [hunk 1]
```diff
@@ -76,36 +78,42 @@ func (jr *JobReceiver) handleJob() http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
 
-		log.Println("Simulating JobReceiver producing a message")
+		principal, _ := middlewares.GetPrincipal(req.Context())
```
Comment (Jason-RH) [inline internal/controller/api/job_receiver.go:81]: Just a note, but I probably should have put the auth middleware into its own package.

File: internal/controller/api/management.go [hunk 2]
```diff
@@ -203,8 +222,15 @@ func (s *ManagementServer) handleConnectionListing() http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
 
+		principal, _ := middlewares.GetPrincipal(req.Context())
+		requestId := request_id.GetReqID(req.Context())
+		logger := logger.Log.WithFields(logrus.Fields{
+			"account":    principal.GetAccount(),
+			"request_id": requestId})
+
+		logger.Debugf("Getting connection list")
```
Comment (Jason-RH) [inline internal/controller/api/management.go:231]: Did you mean to add some variables into this message?

File: internal/controller/api/job_receiver.go (hunk 1 above)
Comment (dehort) [reply internal/controller/api/job_receiver.go:81]: We can always move code around later.

File: internal/controller/api/management.go (hunk 2 above)
Comment (dehort) [reply internal/controller/api/management.go:231]: I'm basically just logging that we received a request, who the request was made by (account field) and the request_id.

**Pull request merged**